src/mlpack/methods/ann/layer/fast_lstm.hpp

```diff
@@ -164,6 +164,12 @@ class FastLSTM
   //! Get the number of output units.
   size_t OutSize() const { return outSize; }
 
+  //! Get the size of the weight matrix.
+  size_t WeightSize() const 
+  {
+    return 4 * outSize * inSize + 4 * outSize + 4 * outSize * outSize;
```

Comment:
`weights.n_elem` should work as well, what do you think?

Reply:
Or wait we use it for initialization.

Reply:
The idea is to be able to call `WeightSize()` when `weights` is not initialized, so we should stick with this implementation I think. :+1:  In fact, if we wanted, we could do `weights.set_size(WeightSize(), 1)` for weight initialization in many of the layers.  But I haven't thought through whether there would be any issues with that.

Reply:
(oops, I commented without seeing the second response, sorry! :))

Reply:
@UtR491 do you think you could also modify `fast_lstm_impl.hpp` so that when `weights` is allocated, it uses `weights.set_size(WeightSize())`?  That way, the code to set the weight size is only in one place. :+1:

Reply:
@rcurtin I did [this](https://github.com/mlpack/mlpack/blob/8b3b5522494e9de1f4b01613c0351ef928ecebf5/src/mlpack/methods/ann/layer/fast_lstm_impl.hpp#L45), should I return a std::pair?